### PR TITLE
132 update to hyper 12 

### PR DESCRIPTION
**Pull request by jrconlin**
Closes https://github.com/mozilla-services/autopush-rs/issues/132

### CODE REVIEW

**Comment by fzzzy** [comment]
Wonderful! Thanks so much for taking this on! It's looking great. Is there anything else that needs to be done or can it be marked non-draft?

**Comment by jrconlin** [comment]
I am hoping that things should be reasonably good to go. There are some questions about how it integrates with the general state machine, and I was getting a couple of local test fails. I'd like to take another go over this code today just to double check stuff, but hope to have this out today.

File: autopush/src/megaphone.rs [hunk 1]
```diff
@@ -196,6 +195,7 @@ impl BroadcastChangeTracker {
         }
 
         // Check to see if this broadcast has been updated since initialization
+        #[allow(clippy::iter_nth_zero)] // TODO: replace with non .nth()
```
Comment (Mcat12) [inline autopush/src/megaphone.rs:198]: Using `next()` should suffice

File: autopush-common/Cargo.toml [hunk 2]
```diff
@@ -13,34 +13,34 @@ edition = "2018"
...
-chrono = "0.4.10"
+cadence = "0.20.0"
+chrono = "0.4.11"
 config = "0.10.1"
 error-chain = "0.12.2"
 # XXX: pin to 0.1 until likely hyper 0.13
 futures = "0.1.29"
 futures-backoff = "0.1.0"
 httparse = "1.3.4"
 # XXX: pin to hyper 0.11 for now: 0.12 has many changes..
```
[the remark this answers is not in the document]
Comment (jrconlin) [reply autopush-common/Cargo.toml]: Yeah, some of these we'll have to punt on until hyper 13. Noted those.

File: autopush/src/http.rs [hunk 3]
```diff
@@ -7,77 +7,87 @@
...
+                                Body::from("Client not available.")
+                            };
+                            Ok(response.body(body).unwrap())
+                        }))
                     } else {
-                        Ok(hyper::Response::new()
-                            .with_status(hyper::StatusCode::BadRequest)
-                            .with_body("Unable to decode body payload"))
+                        Either::B(ok(response
+                            .status(hyper::StatusCode::NOT_FOUND)
```
Comment (pjenvey) [inline autopush/src/http.rs:66]:
```suggestion
                            .status(hyper::StatusCode::BAD_REQUEST)
```

File: autopush/src/http.rs [hunk 4]
```diff
@@ -7,77 +7,87 @@
...
-                    response.set_status(StatusCode::NotFound);
-                    response.set_body("Client not available.");
-                }
+            (&Method::PUT, "notif", uaid) => {
+                return Box::new(clients.check_storage(uaid).then(move |result| {
+                    let body = if result.is_ok() {
+                        response.status(StatusCode::OK);
+                        Body::empty()
+                    } else {
+                        response.status(StatusCode::BAD_GATEWAY);
```
Comment (pjenvey) [inline autopush/src/http.rs:78]:
```suggestion
                        response.status(StatusCode::NOT_FOUND);
```

File: autopush/src/http.rs (hunk 3 above)
Comment (jrconlin) [reply autopush/src/http.rs:66]: heh, weren't these in your updated PR?

reverting.

**Pull request merged**